syntax = "proto3";

package NAlice;

option go_package = "a.yandex-team.ru/alice/megamind/protos/common";
option java_package = "ru.yandex.alice.megamind.protos.common";

import "mapreduce/yt/interface/protos/extension.proto";

// These parameters identify the source which formed given object
message TOrigin {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    oneof OptionalDeviceId {
        option (NYT.variant_field_name) = "optional_device_id";
        string DeviceId = 1 [json_name = "device_id", (NYT.column_name) = "device_id"];
    }

    oneof OptionalUuid {
        option (NYT.variant_field_name) = "optional_uuid";
        string Uuid = 2 [json_name = "uuid", (NYT.column_name) = "uuid"];
    }
}
